src/find/matchers/time.rs

```diff
@@ -362,4 +504,149 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn newer_option_matcher() {
+        #[cfg(target_os = "linux")]
+        let x_options = ["a", "c", "m"];
+        #[cfg(not(target_os = "linux"))]
+        let x_options = ["a", "B", "c", "m"];
+        #[cfg(target_os = "linux")]
+        let y_options = ["a", "c", "m"];
+        #[cfg(not(target_os = "linux"))]
+        let y_options = ["a", "B", "c", "m"];
+
+        for x_option in &x_options {
+            for y_option in &y_options {
+                let temp_dir = Builder::new().prefix("example").tempdir().unwrap();
+                let temp_dir_path = temp_dir.path().to_string_lossy();
+                let new_file_name = "newFile";
+                // this has just been created, so should be newer
+                File::create(temp_dir.path().join(new_file_name)).expect("create temp file");
+                let new_file = get_dir_entry_for(&temp_dir_path, new_file_name);
+                // this file should already exist
+                let old_file = get_dir_entry_for("test_data", "simple");
+                let deps = FakeDependencies::new();
+                let matcher = NewerOptionMatcher::new(
+                    x_option.to_string(),
+                    y_option.to_string(),
+                    &old_file.path().to_string_lossy(),
+                );
+
+                assert!(
+                    matcher
+                        .unwrap()
+                        .matches(&new_file, &mut deps.new_matcher_io()),
+                    "new_file should be newer than old_dir"
+                );
+
+                // After the file is deleted, DirEntry will point to an empty file location,
+                // thus causing the Matcher to generate an IO error after matching.
+                let _ = fs::remove_file(&*new_file.path().to_string_lossy());
+                let matcher = NewerOptionMatcher::new(
+                    x_option.to_string(),
+                    y_option.to_string(),
+                    &old_file.path().to_string_lossy(),
+                );
+                assert!(
+                    !matcher
+                        .unwrap()
+                        .matches(&new_file, &mut deps.new_matcher_io()),
+                    "The correct situation is that the file reading here cannot be successful."
+                );
+            }
+        }
+    }
+
+    #[test]
+    fn newer_time_matcher() {
+        let deps = FakeDependencies::new();
+        let time = deps
+            .new_matcher_io()
+            .now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap()
+            .as_millis()
+            .try_into()
+            .unwrap();
+
+        let created_matcher = NewerTimeMatcher::new(NewerOptionType::Birthed, time);
+
+        thread::sleep(Duration::from_secs(2));
```

Comment:
same, is this sleep needed?

Reply:
This sleep() is used to ensure that there is a small distance between the time when assert() is executed and the time when the file was created. In fact, this code fails the test on my machine after removing the sleep().

and...
Currently the i64 type microsecond timestamp is used to determine the time. I have noticed that it is not accurate enough and I am going to change it to nanoseconds.

Reply:
two seconds is long, try 0.1 ?

Reply:
`Duration::from_secs()` does not accept floating point arguments, maybe `Duration::from_millis(100)` would also be suitable?